### PR TITLE
unify configs in pyproject & ruff

### PR DESCRIPTION
**Pull request by Borda**
## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## What does this PR do?

unify all python related configs in pyproject.toml
replace flake8 by ruff

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
let's wait the PL update is resolved in #222

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -60,7 +60,11 @@ repos:
           - mdformat-black
           - mdformat_frontmatter
 
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.226
```
Comment (akihironitta) [inline .pre-commit-config.yaml:64]: maybe use the latest release?
```suggestion
    rev: v0.0.233
```

**Review by akihironitta — APPROVED** [review]
LGTM!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Borda) [reply .pre-commit-config.yaml:64]: Sure, it is just moving so fast...

**Pull request merged**